### PR TITLE
feat: custom domain name

### PR DESCRIPTION
**Pull request by jjfrench**
- adds logic to create custom domain names for stac-api, titiler-pgstac-api, and stac-ingestor-api
- for custom domain names to be deployed, the following environment variables must be defined before deployment : `CERTIFICATE_ARN` (the ARN of the certificate to be used for the domain name), `INGESTOR_CUSTOM_DOMAIN_NAME` (if missing, no domain name will be created for the ingestor API), `STAC_API_CUSTOM_DOMAIN_NAME` (if missing, no domain name will be created for the STAC API) and `TITILER_PGSTAC_API_CUSTOM_DOMAIN_NAME` (if missing, no domain name will be created for the titiler pgstac API)
- there must be route53 records pointing to each custom domain names created here. Those records are not created here, so they must be manually updated every time the custom domain names are re-created (because their regional domain name changes).

### CODE REVIEW

**Comment by emileten** [comment]
This was deployed locally to the test environment with the `main` branch of `eoapi-cdk` and it's breaking the titiler pgstac. `https://titiler-pgstac-api.dit.maap-project.org/docs` gives `Internal Server Error`. Logs from the AWS console : 

```
[ERROR] ValidationError: 1 validation error for PostgresSettings
database_url
  can only concatenate str (not "NoneType") to str (type=type_error)
Traceback (most recent call last):
  File "/var/lang/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/var/task/handler.py", line 9, in <module>
    from titiler.pgstac.main import app
  File "/tmp/pip-target-wacn12dk/lib/python/titiler/pgstac/main.py", line 42, in <module>
  File "pydantic/env_settings.py", line 40, in pydantic.env_settings.BaseSettings.__init__
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
```

By locally deploying with `eoapi-cdk` `main` we incorporated this change https://github.com/developmentseed/eoapi-cdk/pull/61, that significantly bumps `titiler-pgstac` and changes a little bit the handler, I strongly suspect this is the cause.

**Comment by emileten** [comment]
I think I found the bug, I am going to try to redeploy to test with a fix right now

File: cdk/PgStacInfra.ts [hunk 1]
```diff
@@ -174,5 +195,28 @@ export interface Props extends StackProps {
...
+   * ARN of ACM certificate to use for CDN.
+   * Example: "arn:aws:acm:us-west-2:123456789012:certificate/12345678-1234-1234-1234-123456789012"
+  */
+  certificateArn?: string | undefined;
+
+  /**
+   * Domain name to use for CDN.
+   * Example: "stac.maap.xyz"
+  */
+  IngestorDomainName?: string | undefined;
```
Comment (emileten) [inline cdk/PgStacInfra.ts:209]: I suggest to highlight that a CDN is _created_ 

`* Domain name to use for CDN. If defined, a new CDN will be created`

Same for the two below.

File: cdk/config.ts [hunk 2]
```diff
@@ -16,6 +20,7 @@ export class Config {
       author: String(process.env.AUTHOR),
       gitCommit : String(process.env.COMMIT_SHA),
       gitRepository: String(process.env.GIT_REPOSITORY),
+      version: String(process.env.VERSION),
```
Comment (emileten) [inline cdk/config.ts:23]: let's remember to address how we version this deployment in another PR.

File: cdk/PgStacInfra.ts [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
 import { Construct } from "constructs";
 import {
   BastionHost,
   PgStacApiLambda,
   PgStacDatabase,
   StacIngestor,
-  TitilerPgstacApiLambda
-} from "cdk-pgstac";
+  TitilerPgstacApiLambda,
+} from "../eoapi-cdk/lib";
```
Comment (emileten) [inline cdk/PgStacInfra.ts:16]: We need to change this to "eoapi-cdk" once the eoapi-cdk PR is merged.

**Review by emileten — CHANGES_REQUESTED** [review]
I was also unable to create the ingestor domain name when deploying locally. I fixed this bug and the other one I mentioned (titiler-pgstac broken) in https://github.com/developmentseed/eoapi-cdk/pull/63. 

I am not approving this yet, as we need to change the import statement once the eoapi-cdk PR is merged.

File: cdk/config.ts (hunk 2 above)
Comment (jjfrench) [reply cdk/config.ts:23]: I used this tag as a way to force cdk to redeploy - saw VEDA was doing something similar with their backend

File: cdk/PgStacInfra.ts (hunk 3 above)
Comment (emileten) [reply cdk/PgStacInfra.ts]: done

**Review by emileten — APPROVED** [review]
deployed on GH to test and dev

**Pull request merged**